src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -541,11 +553,23 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCompilationStarted(
           caller.name, "()");
   }
 
+  // IIS: Ensure that the startup hook is inserted into System.Web.Compilation.BuildManager.InvokePreStartInitMethods.
+  // This will be the first call-site considered for the startup hook injection,
+  // which correctly loads Datadog.Trace.ClrProfiler.Managed.Loader into the application's
+  // own AppDomain because at this point in the code path, the ApplicationImpersonationContext
+  // has been started.
+  auto valid_startup_hook_callsite = true;
+  if (module_metadata->assemblyName == "System"_W ||
```

Comment:
Is this because the target method could be in `System.dll` for some versions of the framework?

Reply:
So there's two things going on here to explain why this extra check is necessary:
1. We track JITCompilationStarted events inside `System.dll` because our current `WebRequest` integration instruments an internal call inside the assembly. For that reason, methods from `System.dll` will run through this code path
1. There are some method calls in `System.dll` that execute before the `System.Web.dll` method that we target for the startup hook insertion, so I figured we could just broadly ignore all `System.dll` methods specifically for the purposes of the startup injection

Reply:
The logic reads a little weird, especially starting with `valid_startup_hook_callsite = true` but if you can confirm my assumption about where the target method is, that would be good.

Reply:
Ok, great! So with web apps then, we know we're not going to put the startup hook in `System`. Thanks for explaining.